### DISCUSSION TITLE
Make fields "maskable"

### DISCUSSION THREAD

**Discussion by niels-bosman** [laravel/nova-issues#4550] [Ideas & Feature Requests] [answered]
Hey,

I noticed that since v4.0 Nova Text fields are now copyable by using the `copyable()` method. Maybe text fields could also include a `mask()` method to hide the actual contents of the field and show * instead (like https://github.com/sixlive/nova-text-copy-field). This could allow users to use passwords in an application in a good way.

**Comment by crynobone** [comment]
> This could allow users to use passwords in an application in a good way.

The XHR requests still need to download the "password"/"key" in plain text. Which exposes it to any Vue component that may have access to the field response object.

**Reply by niels-bosman** [reply]
I get that, but I think there are cases where you would want to be able to share a "password" / "key" with users of a system while not making it visible. For example, we use Nova to manage our company passwords. Making the text masked will in that case strongly decline the chance of any password being leaked by for example someone in office taking a picture of the screen.

Another option would be to make the text that's copyable not visible but only show the icon, but I do not think that is possible. Is it?

**Comment by ThisIsMyFavouriteJar** [comment]
This is similar to https://github.com/laravel/nova-issues/discussions/4230

Basically you do not always want the text to be copied to be the same as the displayed text.
I want to copy some instructions, and they take up a lot of space if not truncated.

Same for the password / key situation described, you just want the displayed text and the copied text to be different.

Hope this can be added!

**Reply by niels-bosman** [reply, marked as answer]
I managed to get it fixed for my situation by using the `displayUsing()` chainable method.

It looks like this in my case.

```php
Text::make('Password')
      ->rules('required')
      ->displayUsing(fn (): string => ' ')
      ->copyable(),
```

Unfortunately though, this callback that you need to pass to get to a value must NOT be empty. So in my case I had to insert one space.

You could make the callback return anything you like though in your case.

**Reply by ThisIsMyFavouriteJar** [reply]
Hi Niels, this is exactly what I was looking for! Bedankt!
